cmds/admin_server/job/sql/stroage.go

```diff
@@ -0,0 +1,102 @@
+package sql
+
+import (
+	"fmt"
+
+	"github.com/google/go-safeweb/safesql"
+	jobstorage "github.com/linuxboot/contest/cmds/admin_server/job"
+	"github.com/linuxboot/contest/pkg/xcontext"
+)
+
+const (
+	tagsStmt = `SELECT tag, COUNT(tag) FROM job_tags WHERE tag REGEXP CONCAT('.*',?,'.*') GROUP BY tag`
+	jobStmt  = `SELECT t.job_id, r.reporter_name, r.report_time, r.data FROM job_tags t LEFT JOIN final_reports r ON t.job_id = r.job_id WHERE t.tag = ?`
+)
+
+// SQL defines a struct that wraps a db connection to job sql database
+type Storage struct {
+	db *safesql.DB
+}
+
+func New(dbURI, driveName string) (*Storage, error) {
+
+	if driveName == "" {
+		driveName = "mysql"
+	}
+
+	db, err := safesql.Open(driveName, dbURI)
+	if err != nil {
+		return nil, fmt.Errorf("error while initializing the db: %w", err)
+	}
+	if err := db.Ping(); err != nil {
+		db.Close()
+		return nil, fmt.Errorf("error while connecting to the db: %w", err)
+	}
+
+	return &Storage{
+		db: &db,
+	}, nil
+}
+
+// GetTags returns project/tags that has a tag matches tagPattern
```

Comment:
this code here really has no concept of "project". youre just handling tags. if you wanna make this about projects, make it explicit in the naming of the functions, structs, etc
```suggestion
// GetTags returns tags that has a tag matches tagPattern
```